pkg/targets/delegation.go

```diff
@@ -57,7 +72,7 @@ func (d *delegationsIterator) Next() (value Delegation, ok bool) {
 	return delegation, true
 }
 
-func (d *delegationsIterator) Add(roles []data.DelegatedRole, delegator string, verifier verify.DelegationsVerifier) error {
+func (d *delegationsIterator) Add(roles []data.DelegatedRole, delegator string, db *verify.DB) error {
```

Comment:
A delegation (an edge in the traversal) is defined in part by a set of keys (see [spec](https://theupdateframework.github.io/specification/latest/#delegations)), so the keys DB is specific to particular edge. It would not be correct to use the same DB for the entire traversal.

Reply:
I see. I can not make specific suggestion but I wish there was a way this code was more intuitive. Other than that, this looks great!